tests/unit/backend/gene_info/test_v1.py

```diff
@@ -49,20 +49,43 @@ def test_api_calls(self, mock_load_search_result, mock_get):
     @patch("backend.gene_info.api.v1.NCBIProvider.fetch_gene_uid")
     @patch("backend.gene_info.api.v1.NCBIProvider.fetch_gene_info_tree")
     @patch("backend.gene_info.api.v1.NCBIProvider.parse_gene_info_tree")
-    def test_fetches_and_searches(self, mock_parse_gene_info_tree, mock_fetch_gene_info_tree, mock_fetch_gene_uid):
+    @patch("backend.gene_info.api.v1.GeneChecker.get_id")
+    def test_fetches_and_searches(
+        self, mock_get_id, mock_parse_gene_info_tree, mock_fetch_gene_info_tree, mock_fetch_gene_uid
+    ):
         """
         Successfully calls NCBIProvider fetch and search functions with correct parameters
         """
         mock_fetch_gene_uid.return_value = 1
         mock_fetch_gene_info_tree.return_value = None
         mock_parse_gene_info_tree.return_value = self.final_gene_info_result
+        mock_get_id.return_value = "ensembl1"
+
+        # parameters contain only gene ID
         res = self.app.get("/gene_info/v1/gene_info?geneID=ensembl1")
         self.assertTrue(mock_fetch_gene_uid.called)
         self.assertTrue(mock_fetch_gene_info_tree.called)
         self.assertTrue(mock_parse_gene_info_tree.called)
         self.assertEqual(res.status_code, requests.codes.ok)
         self.assertEqual(json.loads(res.data), self.final_gene_info_result)
 
+        # parameters contain both gene ID and gene name
```

Comment:
re my comment above you could probably just check that mock_get_id wasnt called.
Also I think you'll want to either reset the mocks or pull this into a separate test to avoid incorrect results. I'd recommend creating a separate test